### ISSUE TITLE
fatal runtime error: stack overflow

### ISSUE THREAD

**Issue by jazsakr** [nanoporetech/modkit#44]
Hello!

I am using `modkit extract` on my modified bam files that were basecalled with guppy v6.5.7. I have two samples, one job completed (I will call it bam1) successfully and for the other sample (I will call bam2), I got a `fatal runtime error: stack overflow` error. 

This is the command I ran: `modkit extract bl6_10F_03_methyl.bam bl6_10F_03_methyl_extract.tsv --reference ~/references/mm39.fa --include-bed ~/references/mm39_cg_motifs.bed --threads 8`

And it gave me the following output and error:

```
> specifying include-only BED outputs only mapped sites
> parsing BED at /home/jsakr/references/mm39_cg_motifs.bed
> processed 43845398 BED lines
▹▹▹▹▹ [00:41:08] 0 ~records failed
▹▹▹▹▹ [00:41:08] 116138 ~records skipped
▸▹▹▹▹ [00:41:08] 751607 ~records used
▹▹▹▹▹ [00:41:08] 103214646 rows written
[00:00:00] ----------------------------------------       0/61      contigs
[00:01:45] ##############--------------------------     596/1818    processing chr2 
thread '<unknown>' has overflowed its stack
fatal runtime error: stack overflow
Aborted (core dumped)
```

I tried to run this same command for bam2 on different machines and got the same error. I also tried increase the number of threads to 16. I added `--log-filepath` and this is the last few lines of the log file:

```
[src/mod_bam.rs::63][2023-08-11 17:44:41][DEBUG] record b66026b6-18a9-44f1-b72f-f4fbee839b67 has no base modification information, skipping
[src/mod_bam.rs::63][2023-08-11 17:44:42][DEBUG] record ee1287f1-ac56-49a2-afef-5e991fc16723 has no base modification information, skipping
[src/mod_bam.rs::63][2023-08-11 17:44:46][DEBUG] record 616c20a1-4c76-4408-a3d5-71bd8f028a36 has no base modification information, skipping
[src/mod_bam.rs::63][2023-08-11 17:45:20][DEBUG] record 5c46ab57-0bd3-427b-af46-1aa90fc8e6c6 has no base modification information, skipping
[src/mod_bam.rs::63][2023-08-11 17:45:21][DEBUG] record 5c46ab57-0bd3-427b-af46-1aa90fc8e6c6 has no base modification information, skipping
```

A difference between the two files is that bam1 is much smaller than bam2 (4.2 Gbases vs 14.9 Gbases of data). I am not sure how significant this is.  Any suggestions? 

Thank you!

Jaz

**Comment by ArtRand** [comment]
Hello @jazsakr,

I have a hypothesis, do you see many of these debug lines "..has no base modification information, skipping"? Could you remove unmapped reads from bam2?

**Comment by ArtRand** [comment]
Hello @jazsakr 

I've made a build that I think will fix this issue. Find it in the [releases section](https://github.com/nanoporetech/modkit/releases/tag/v0.1.13-rc1). Please let me know if it fixes the problem.

**Comment by jazsakr** [comment]
Hello! 

Thank you so much for the quick response!

With the old version (v0.1.12), I tested your theory. I removed all the unmapped reads and tried again. Unfortunately, the same error happened again. To check, I looked up some of the read IDs from the modkit log to the bam file and they are indeed mapped reads. I installed the new version (v0.1.13-rc1) and tried it on the unfiltered bam file. It worked! 

Thank you so much!

Jaz
